src/ProductConstructionService/ProductConstructionService.DependencyFlow/PullRequestBuilder.cs

```diff
@@ -211,15 +218,44 @@ public async Task<string> GenerateCodeFlowPRDescriptionAsync(SubscriptionUpdateW
             This pull request is bringing source changes from **{update.SourceRepo}**.
             
             - **Subscription**: {update.SubscriptionId}
-            - **Build**: {build.AzureDevOpsBuildNumber}
+            - **Build**: [{build.AzureDevOpsBuildNumber}]({build.GetBuildLink()})
             - **Date Produced**: {build.DateProduced.ToUniversalTime():MMMM d, yyyy h:mm:ss tt UTC}
-            - **Commit**: {build.Commit}
+            - **Source Diff**: {sourceDiffText}
+            - **Commit**: [{build.Commit}]({build.GetCommitLink()})
             - **Branch**: {build.GetBranch()}
 
             {GetEndMarker(update.SubscriptionId)}
             """;
     }
 
+    private string CreateSourceDiffLink(
+        SubscriptionUpdateWorkItem update,
+        Microsoft.DotNet.ProductConstructionService.Client.Models.Build build,
+        string previousSourceCommit)
+    {
+        if (string.IsNullOrEmpty(previousSourceCommit))
+        {
+            return COMMIT_DIFF_NOT_AVAILABLE_TXT;
+        }
+
+        string shortPrevCommitSha = previousSourceCommit.Length >= 7 ? previousSourceCommit.Substring(0, 7) : previousSourceCommit;
+        string shortBuildCommitSha = build.Commit.Length >= 7 ? build.Commit.Substring(0, 7) : build.Commit;
```

Comment:
There is class called `Commit` that has a method for this already.
Also I think we know the length will always be the same (and more than 7).